src/Microsoft.Azure.WebJobs.Logging/FunctionInstanceLogItem.cs

```diff
@@ -153,6 +160,57 @@ public void Validate()
                 }
             }
         }
+                     
+        /// <summary>
+        /// Truncate various fields to fit in logging sizes. 
+        /// </summary>
+        public void Truncate()
+        {
+            // This is fundamentally driven by performance. We need to fit log entries into table rows.
+            // Truncate to ensure that we're under table's maximum request payload size (4mb).
+            // None of these limits (except for output log)  should actually get hit in normal scenarios. 
+            this.TriggerReason = Truncate(this.TriggerReason, MaxTriggerReasonLength);
+            this.ErrorDetails = Truncate(this.ErrorDetails, MaxErrorLength);
+
+            // Logger should already have truncated this, but just in case. 
+            this.LogOutput = Truncate(this.LogOutput, MaxLogOutputLength);
+
+            // Arguments may have 1 larger argument for the trigger. 
```

Comment:
I think for a large number of arguments, I'll just cut the whole thing.  This shouldn't happen in practice. 
I think the common case is 1 (potentially large) argument from the trigger; and several smaller ones (like output bindings).